### ISSUE TITLE
Consider using `define-enumeration` for nullary sum types

### ISSUE THREAD

**Issue by JordanMartinez** [purescm/purescm#124]
For types like this:
```purs
data Foo
  = Bar
  | Baz
  | Wat
```

Maybe we should use [`define-enumeration`](https://scheme.com/tspl4/objects.html#./objects:h14) rather than `define-record-type`?

**Comment by purefunctor** [comment]
Hmm, I'm not sure. After reading through initially, my understanding is that enumerations are designed to be similar in use to C-style enums, in that its primary purpose is for providing flag-like behavior.

**Comment by JordanMartinez** [comment]
Yeah, I forgot that pattern matching is involved as well, which `define-enumeration` doesn't handle as well as just using symbols does.
